src/Microsoft.Health.Fhir.Shared.Api/Features/Exceptions/BaseExceptionMiddleware.cs

```diff
@@ -105,13 +112,30 @@ public async Task Invoke(HttpContext context)
                     operationOutcome,
                     exception is ServiceUnavailableException ? HttpStatusCode.ServiceUnavailable : HttpStatusCode.InternalServerError);
 
+                doesOperationOutcomeHaveError = true;
+
                 await ExecuteResultAsync(context, result);
             }
+            finally
+            {
+                EmitHttpFailureMetricInCaseOfError(context, doesOperationOutcomeHaveError);
+            }
         }
 
         protected internal virtual async Task ExecuteResultAsync(HttpContext context, IActionResult result)
         {
             await result.ExecuteResultAsync(new ActionContext { HttpContext = context });
         }
+
+        private void EmitHttpFailureMetricInCaseOfError(HttpContext context, bool doesOperationOutcomeHaveError)
+        {
+            if (context.Response.StatusCode >= 500 || doesOperationOutcomeHaveError)
+            {
+                string operationName = context.Request.GetOperationName(includeRouteValues: false);
+
+                _failureMetricHandler.EmitHttpFailure(
```

Comment:
Does it make sense to have additional info in this metric like HTTP code? Or since they are all >= 500 perhaps it won't tell us much? 

I'm thinking we way want to measure 503s vs 500s but not sure how this is going to be used downstream.

Reply:
Yeap. I had the same thinking when developing this. 

The current metric tracks HTTP failures (>=500).
I think that from the DRI perspective, if a high volume of 5**s are happening, you don't care if they are 500, 503 or 504. 
It should probably act fast, because it's a problem on our side. 

That was my line of thought.

Reply:
Makes sense - we can always get more info from logs. 👍🏽